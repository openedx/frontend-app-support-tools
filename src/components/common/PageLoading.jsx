import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class PageLoading extends Component {
  renderSrMessage() {
    if (!this.props.srMessage) {
      return null;
    }

    return <span className="sr-only">{this.props.srMessage}</span>;
  }

  render() {
    return (
      <div>
        <div
          data-testid="page-loading"
          className="d-flex justify-content-center align-items-center flex-column"
          style={{
            height: '20vh',
          }}
        >
          <div className="spinner-border text-primary" role="status">
            {this.renderSrMessage()}
          </div>
        </div>
      </div>
    );
  }
}

PageLoading.propTypes = {
  srMessage: PropTypes.string.isRequired,
};
