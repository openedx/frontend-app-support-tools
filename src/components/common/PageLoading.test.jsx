import React from 'react';
import { render, screen } from '@testing-library/react';
import PageLoading from './PageLoading';
import '@testing-library/jest-dom';

describe('<PageLoading />', () => {
  it('does not render screen reader message when srMessage is empty', () => {
    render(<PageLoading srMessage="" />);
    const srElement = screen.queryByText((content, element) => element?.classList.contains('sr-only'));
    expect(srElement).not.toBeInTheDocument();
  });

  it('renders expected message', () => {
    const message = 'Loading...';
    render(<PageLoading srMessage={message} />);
    const srElement = screen.getByText(message);
    expect(srElement).toBeInTheDocument();
    expect(srElement).toHaveClass('sr-only');
  });

  it('snapshot matches correctly', () => {
    const { asFragment } = render(<PageLoading srMessage="Loading" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
