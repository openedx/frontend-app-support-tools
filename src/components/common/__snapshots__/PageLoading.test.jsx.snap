// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLoading /> snapshot matches correctly 1`] = `
<div>
  <div
    className="d-flex justify-content-center align-items-center flex-column"
    style={
      {
        "height": "20vh",
      }
    }
  >
    <div
      className="spinner-border text-primary"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading
      </span>
    </div>
  </div>
</div>
`;
